File: Buoi2/Julia/Insertion Sort.jl
# Hàm Insertion Sort
function insertion_sort(arr)
    n = length(arr)
    for i in 2:n
        key = arr[i]  # Lấy phần tử cần chèn
        j = i - 1  # Tìm vị trí chèn đúng

        # Di chuyển các phần tử lớn hơn key sang phải
        while j >= 1 && arr[j] > key
            arr[j + 1] = arr[j]
            j -= 1
        end
        arr[j + 1] = key  # Chèn key vào vị trí đúng
    end
end

arr = [5, 2, 9, 1, 5]
insertion_sort(arr)
println("Kết quả sau Insertion Sort:", arr)

File: Buoi2/C++/Insertion Sort.cpp
#include <iostream>
#include <vector>
using namespace std;

// Hàm Insertion Sort
void insertionSort(vector<int>& arr) {
    int n = arr.size();
    for (int i = 1; i < n; i++) {
        int key = arr[i]; // Lấy phần tử cần chèn
        int j = i - 1; // Tìm vị trí chèn đúng

        // Di chuyển các phần tử lớn hơn key sang phải
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = key; // Chèn key vào vị trí đúng
    }
}

int main() {
    vector<int> arr = {5, 2, 9, 1, 5};
    insertionSort(arr);
    cout << "Ket qua sau Insertion Sort: ";
    for (int x : arr) {
        cout << x << " ";
    }
    cout << endl;
    return 0;
}
